tests/helpers.bash

```diff
@@ -341,7 +341,9 @@ function assert() {
             return
         fi
     else
-        if eval "[[ $not \$actual_string $actual_op \$expect_string ]]"; then
+        # Escaping the strings with quotes is important, otherwise we run
+        # into the same problem as described in the comment above.
+        if eval "[[ $not \"\$actual_string\" $actual_op \"\$expect_string\" ]]"; then
```

Comment:
Preference for:
```diff
index 2bb6b429..1be49b03 100644
--- a/tests/helpers.bash
+++ b/tests/helpers.bash
@@ -340,6 +340,11 @@ function assert() {                                                                                                                       
         if [ "$actual_string" = "$expect_string" ]; then
             return
         fi
+    elif [[ $operator == '!=' ]]; then
+        # Same special case as above
+        if [ "$actual_string" != "$expect_string" ]; then
+            return
+        fi
     else
         if eval "[[ $not \$actual_string $actual_op \$expect_string ]]"; then
             return
```
Reason: the quoting-approach (which is what I initially suggested, my bad) will fail completely when operator is `~` (i.e. when we really _do_ want patterns).

Reply:
added, thanks